src/config/__tests__/settings.spec.ts

```diff
@@ -0,0 +1,228 @@
+      },
+      portForwarding: { includeKubernetesServices: false },
+      images:         {
+        showAll:   true,
+        namespace: 'k8s.io'
+      },
+      telemetry:              true,
+      updater:                true,
+      debug:                  true,
+      pathManagementStrategy: PathManagementStrategy.NotSet,
+    };
+    origPrefs = JSON.parse(JSON.stringify(prefs));
+  });
+
+  describe('getUpdatableNode', () => {
+    test('returns null on an invalid top level accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'blah-blah-blah');
+
+      expect(result).toBeNull();
+    });
+    test('returns null on an invalid internal accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'kubernetes-options-blah');
+
+      expect(result).toBeNull();
+    });
+    test('returns the full pref with a top-level accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'kubernetes') as [Record<string, any>, string];
+
+      expect(result).not.toBeNull();
+      const [lhs, accessor] = result;
+
+      expect(lhs).toEqual(prefs);
+      expect(accessor).toBe('kubernetes');
+    });
+    test('returns a partial pref with an internal accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'kubernetes-options-flannel') as [Record<string, any>, string];
+
+      expect(result).not.toBeNull();
+      const [lhs, accessor] = result;
+      const flannelNow = prefs.kubernetes.options.flannel;
+      const flannelAfter = !flannelNow;
+
+      expect(lhs).toEqual({
+        traefik: prefs.kubernetes.options.traefik,
+        flannel: flannelNow,
+      });
+      expect(accessor).toBe('flannel');
+      lhs[accessor] = flannelAfter;
+      expect(prefs.kubernetes.options.flannel).toBe(flannelAfter);
+    });
+  });
+
+  test('no command-line args should leave prefs unchanged', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, []);
+
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('one option with embedded equal sign should change only one value', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, ['--kubernetes-version=1.23.6']);
+
+    expect(newPrefs.kubernetes.version).toBe('1.23.6');
+    newPrefs.kubernetes.version = origPrefs.kubernetes.version;
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('one option over two args should change only one value', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, ['--kubernetes-version', '1.23.7']);
+
+    expect(newPrefs.kubernetes.version).toBe('1.23.7');
+    newPrefs.kubernetes.version = origPrefs.kubernetes.version;
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('boolean option to true should change only that value', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, ['--kubernetes-options-flannel=true']);
+
+    expect(newPrefs.kubernetes.options.flannel).toBeTruthy();
+    newPrefs.kubernetes.options.flannel = false;
```

Comment:
```suggestion
    newPrefs.kubernetes.options.flannel = origPrefs.kubernetes.options.flannel;
```
(To make it clearer that we resetting the value, rather than setting it to something arbitrary.)

This also applies to the next two tests… the others are already in this style.

Reply:
Note that these tests assume the specified value in the assignment to `prefs`, rather than paying attention to the default values (or overriding them).

I also put assertions in of what we're expecting the original pref to be, and doing that after `updateFromCommandLine` is invoked to show that the original and new settings are different.

Reply:
Right; they're not using `defaultSettings`, but they're using `origPrefs`.  At least, they're doing that in order to match on the next line where we check they equal `origPrefs`.